src/main/java/request/HttpRequestParser.java

```diff
@@ -0,0 +1,31 @@
+package com.td.HttpServer;
+import java.util.*;
+
+public class HttpRequestParser implements IRequestParser {
+
+  private IValidator requestVerifier;
+  public HttpRequestParser(IValidator requestVerifier) {
+    this.requestVerifier = requestVerifier;
+  }
+
+  public String parseRequestLine(String rawRequest) throws InvalidHttpRequestException {
+    String requestLine = rawRequest.split("\r?\n")[0];
+    if (!requestVerifier.isValid(requestLine)) {
```

Comment:
Hmmm... I wonder if this `requestVerifier` should still be injected. Maybe it could be pulled into this class? Or at least, maybe it can be just used directly by this class, no need to inject it?